flyteadmin/pkg/repositories/transformers/task_execution.go

```diff
@@ -308,16 +310,44 @@ func mergeExternalResource(existing, latest *event.ExternalResourceInfo) *event.
 	if latest.CacheStatus != core.CatalogCacheStatus_CACHE_DISABLED && existing.CacheStatus != latest.CacheStatus {
 		existing.CacheStatus = latest.CacheStatus
 	}
+	if latest.CatalogKey != nil {
+		existing.CatalogKey = latest.CatalogKey
+	}
 	existing.Logs = mergeLogs(existing.Logs, latest.Logs)
 
-	return existing
+	// set output metadata if exists
+	if len(latest.GetOutputUri()) > 0 || latest.GetError() != nil {
```

Comment:
would we want to add a check if existing.OutputResult is not set to avoid making the uri, err := common.OffloadLiteralMap call multiple times?

I know above there's checks if existing.Value != latest.Value, but that wouldn't be applicable due to offloaded outputs needing to be fetched to check values. If the outputResult is a permanent/non-changing state, then the existing.OutputResult != nil should be equivalent.

Reply:
Right, I was relying on the fact this is only ever reached once e2e. But probably better to be explicit about this as you mentioned. Thanks!